server/src/pages/shot/highlighter-tool.js

```diff
@@ -0,0 +1,115 @@
+
+let previousPosition;
+let points = [];
+
+exports.HighlighterTool = class HighlighterTool extends DrawingTool {
+  constructor(props) {
+    super(props);
+    this.state = {
+      strokeStyle: props.color || "#000",
+      lineWidth: props.lineWidth || 20,
+      classNames: "highlighter"
+    };
+    this.canvasWidth = parseInt(props.baseCanvas.style.width, 10);
+    this.canvasHeight = parseInt(props.baseCanvas.style.height, 10);
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return {
+      strokeStyle: nextProps.color,
+      lineWidth: nextProps.lineWidth,
+      classNames: getClassNamesByColor(nextProps.color)
+    };
+  }
+
+  setDrawingProperties() {
+    this.drawingContext.strokeStyle = this.state.strokeStyle;
+    this.drawingContext.lineWidth = this.state.lineWidth;
+    this.drawingContext.lineCap = "square";
+    this.drawingContext.lineJoin = "round";
+  }
+
+  draw(position) {
+    if (!previousPosition) {
+      previousPosition = position;
+    }
+    points.push(position);
+
+    if (points.length < 3) {
+      this.drawingContext.beginPath();
+      this.drawingContext.moveTo(previousPosition.x, previousPosition.y);
+      this.drawingContext.lineTo(position.x, position.y);
+      this.drawingContext.stroke();
+      previousPosition = position;
+      return;
+    }
+
+    this.drawingContext.moveTo(previousPosition.x, previousPosition.y);
+    this.drawingContext.lineTo(position.x, position.y);
+    this.drawingContext.clearRect(0, 0, this.canvasWidth, this.canvasHeight);
+    this.drawingContext.beginPath();
+    this.drawingContext.moveTo(points[0].x, points[0].y);
+    let i;
+    for (i = 1; i < points.length - 2; i++) {
+      const endX = (points[i].x + points[i + 1].x) / 2;
+      const endY = (points[i].y + points[i + 1].y) / 2;
+      this.drawingContext.quadraticCurveTo(points[i].x, points[i].y, endX, endY);
+    }
+    this.drawingContext.quadraticCurveTo(
+      points[i].x,
+      points[i].y,
+      points[i + 1].x,
+      points[i + 1].y
+    );
+    this.drawingContext.stroke();
+    previousPosition = position;
+  }
+
+  finalize() {
+    const composite = document.createElement("canvas");
+    composite.width = this.props.baseCanvas.width;
+    composite.height = this.props.baseCanvas.height;
+    const compositeContext = composite.getContext("2d");
+    compositeContext.scale(this.props.devicePixelRatio, this.props.devicePixelRatio);
+    compositeContext.drawImage(this.props.baseCanvas,
+      0, 0, this.canvasWidth, this.canvasHeight);
+
+    if (isWhite(this.state.strokeStyle)) {
+      compositeContext.globalCompositeOperation = "soft-light";
+    } else {
+      compositeContext.globalCompositeOperation = "multiply";
+    }
+
+    compositeContext.drawImage(this.canvas.current,
+      0, 0, this.canvasWidth, this.canvasHeight);
+    this.drawingContext.drawImage(composite,
+      0, 0, this.canvasWidth, this.canvasHeight);
+  }
```

Comment:
This is necessary for the highlighter because it doesn't use the default `globalCompositeOperation`.  A complete image is created here for the editor.  (Alternatively we could pass a `globalCompositeOperation` back to the editor.  I prefer to keep editor simpler if possible.)

Code review note: since the affected area is known, the `drawImage()` calls should use that.  It would require a change in `DraingTool` to adjust the affected area for the `lineWdith` before calling this method.